packages/foundation-models/src/azure-openai/stream/line-decoder.ts

```diff
@@ -0,0 +1,105 @@
+type Bytes = string | ArrayBuffer | Uint8Array | Buffer | null | undefined;
+
+/**
+ * A re-implementation of httpx's `LineDecoder` in Python that handles incrementally
+ * reading lines from text.
+ *
+ * Https://github.com/encode/httpx/blob/920333ea98118e9cf617f246905d7b202510941c/httpx/_decoders.py#L258.
```

Comment:
Is there no good JS lib that does this already?

Reply:
Not that I know, maybe @deekshas8 you have seen something like this? I mean I suppose OpenAI implement this due to some good reason.

Reply:
I am only aware of this https://nodejs.org/api/readline.html (this handles the line endings). It expects a stream as input though. I am also not sure if it handles any other custom things that we might be doing in LineDecoder.

Reply:
In this PR, I would save the effort since it works and not touch this fundamental part.